### DISCUSSION TITLE
How to change the default column selection of d3fc charts in perspective

### DISCUSSION THREAD

**Discussion by Madasamy26** [perspective-dev/perspective#1369] [Q&A]
Hi,

If I select the any views from selection dropdown, It will showing default selected columns for specified charts, So can we decide them to which chart has to display which column as default ?

**Comment by texodus** [comment]
`<perspective-viewer>` tries to preserve the current selected columns, with the caveat that plugins may specify reasonable bounds for defaults.  This is hard to translate when switching between datagrid and charts, as a e.g. bar chart with all columns is not only generally unreadable, but can overwhelm the browser if too much data is rendered (though perspective has heuristic protection for this also).

You can of course always pre-configure`<perspective-viewer>` using the attribute API, or implement UI buttons outside the viewer which use `restore()` or said attribute API to set the viewer's state to a canned configuration with any settings you'd like (ala the hover buttons on Perspective's [project page](https://perspective.finos.org/).

The actual logic for this is a bit [complex](https://github.com/finos/perspective/blob/master/packages/perspective-viewer/src/js/viewer/dom_element.js#L429).  I'm open to making this more flexible/configurable, but we are in the midst of an overhaul of the plugin API at the moment that will likely change the story a bit, so I'll re-comment when this work lands.
